### PR TITLE
corrected some typos to make the code clean and added two macros

### PR DESCRIPTION
**Pull request by amsorrytola**
1. in complex.huff the add function should take 4 values but it was written take(0) similarly it should be written return(2). It doesnt affect the working of code . 
2. added an function in complex.huff that takes a complex number and angle by which it should rotate it and returns the rotated complex number.
3. added another function that takes two complex numbers as input and returns the equation of a line that passes through it.

### CODE REVIEW

**Comment by 0xpanicError** [comment]
Hey @mohammed-talha-ansari 
Thanks for contributing and working on this repository. Can you also add some unit tests for the function you wrote to ensure that it is implemented correctly? You can refer the tests written in `src/test`
Thanks!

File: src/ComplexHuff/Complex.huff [hunk 1]
```diff
@@ -5,7 +5,7 @@
 
 #define constant VALUE_LOCATION = FREE_STORAGE_POINTER()
 
-#define macro ADD_Z() = takes (0) returns (0) {
+#define macro ADD_Z() = takes (4) returns (0) {
```
Comment (0xpanicError) [inline src/ComplexHuff/Complex.huff:8]: @blueh4mster can you ensure that this change is correct and doesn't introduce any bugs?

**Review by blueh4mster — COMMENTED** [review]
This change doesn't affect macro implementation due to the stack being the same but logically, macro definition should contain 4 instead of 0.

**Comment by 0xpanicError** [comment]
> This change doesn't affect macro implementation due to the stack being the same but logically, macro definition should contain 4 instead of 0.

Then shouldn't it return 2 values as well? Also is this trend followed in all functions?

**Comment by blueh4mster** [comment]
> > This change doesn't affect macro implementation due to the stack being the same but logically, macro definition should contain 4 instead of 0.
> 
> Then shouldn't it return 2 values as well? Also is this trend followed in all functions?

Yes, like I said what happens in the macro definition doesn't affect the actual working of the function but for clean and quality code writing, it is indeed a typo and should be `takes(4) returns(2)`.
As for the other functions, I've checked that they don't have this typo.

**Comment by 0xpanicError** [comment]
> > > This change doesn't affect macro implementation due to the stack being the same but logically, macro definition should contain 4 instead of 0.
> > 
> > 
> > Then shouldn't it return 2 values as well? Also is this trend followed in all functions?
> 
> Yes, like I said what happens in the macro definition doesn't affect the actual working of the function but for clean and quality code writing, it is indeed a typo and should be `takes(4) returns(2)`. As for the other functions, I've checked that they don't have this typo.

Okay. @mohammed-talha-ansari can you also make the above change as mentioned by Preeti in this PR?

**Review by 0xpanicError — APPROVED** [review]

**Pull request closed without merging**